CMakePresets.json

```diff
@@ -0,0 +1,216 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 21,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "nix-base",
+      "hidden": true,
+      "generator": "Ninja",
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER_LAUNCHER": "ccache",
+        "CMAKE_FIND_FRAMEWORK": "LAST",
+        "CMAKE_FIND_PACKAGE_PREFER_CONFIG": "ON",
+        "CMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY": "OFF",
+        "CMAKE_FIND_USE_PACKAGE_REGISTRY": "OFF",
+        "CMAKE_OSX_SYSROOT": "",
+        "VAST_ENABLE_DEVELOPER_MODE": "ON",
+        "VAST_ENABLE_LSVAST": "ON",
+        "VAST_ENABLE_MDX_REGENERATE": "ON",
+        "VAST_PLUGINS": "${sourceDir}/plugins/broker;${sourceDir}/plugins/pcap;${sourceDir}/plugins/sigma;${sourceDir}/plugins/summarize"
+      }
+    },
+    {
+      "name": "clang-base",
+      "hidden": true,
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "clang",
+        "CMAKE_CXX_COMPILER": "clang++"
+      },
+      "environment": {
+        "LDFLAGS": "-fuse-ld=lld"
+      }
+    },
+    {
+      "name": "nix-clang-debug",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Debug Config",
+      "binaryDir": "${sourceDir}/build/clang/debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/debug",
+        "VAST_ENABLE_ASAN": "ON",
+        "VAST_ENABLE_BUNDLED_CAF": "ON",
+        "VAST_PLUGINS": "${sourceDir}/plugins/pcap;${sourceDir}/plugins/sigma;${sourceDir}/plugins/summarize"
+      }
+    },
+    {
+      "name": "nix-clang-redeb",
+      "inherits": ["nix-base", "clang-base"],
+      "displayName": "Clang Redeb Config",
+      "binaryDir": "${sourceDir}/build/clang/redeb",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install/clang/redeb",
+        "VAST_ENABLE_BUNDLED_CAF": "OFF"
```

Comment:
Why do we disable that here? If we enable it for Debug builds because of the added debug information, then we should also enable it for RelWithDebInfo builds for for the added debug information.

Reply:
1. Having a CAF with debug info doesn't add that much value in redeb mode, doing that only for full Debug builds is enough.
2. Switching to Bundled CAF would mean disabling the broker plugin, and then you have to build in Release mode again when you want to check whether the plugin still works.

Reply:
I forgot about that aspect of the Broker plugin. Should we even use the bundled CAF for Debug builds then? It probably makes more sense to just never use it so all presets support every plugin.

Reply:
Hm, I don't know. Being able to jump to definition of CAF function is valuable to me. That doesn't work with prebuilt versions.

Reply:
That is why the debug presets exclude the broker plugin.
Maybe we can use a different method to detect the issue. It should work of the CAF used by broker is the same version as the bundled one.

Reply:
In `plugins/broker/CMakeLists.txt`, change this:

```cmake
if (VAST_ENABLE_BUNDLED_CAF)
  message(FATAL_ERROR "Cannot use bundled CAF with Broker plugin")
endif ()
```

to

```cmake
if (VAST_ENABLE_BUNDLED_CAF)
  message(WARNING "Cannot use bundled CAF with Broker plugin")
  return()
endif ()
```

And move it between the `find_package(VAST)` and `VASTRegisterPlugin(...)` calls. That should already do it.